hatch/src/deps/dependency.rs

```diff
@@ -11,13 +11,17 @@ pub struct Dependency {
 impl Dependency {
   pub fn new(url: String) -> Dependency {
     Dependency {
-      name: (|u: &OsStr| -> String {
-        String::from(u.to_string_lossy())[..].replace(".git", "")
-      })(PathBuf::from(url.clone()).iter().last().unwrap()),
+      name: Dependency::extract_name(url.clone()),
       url,
     }
   }
 
+  fn extract_name(url: String) -> String {
```

Comment:
This function name does not describe what this function is doing. How would I know that it is stripping `.git`?

Reply:
A repo url does not necessarily contain a `.git` extension.  We strip it off iff its there.

Reply:
So make it extension agnostic.

Reply:
What do you mean?